File: PHP/Medium/Sorting/1418__Display_Table_of_Food_Orders_in_a_Restaurant.php
<?php
function displayTable($orders)
{
    $foodItems = [];
    $tableOrders = [];

    foreach ($orders as $order) {
        list($customer, $table, $food) = $order;
        $foodItems[$food] = true;
        if (!isset($tableOrders[$table])) {
            $tableOrders[$table] = [];
        }
        if (!isset($tableOrders[$table][$food])) {
            $tableOrders[$table][$food] = 0;
        }
        $tableOrders[$table][$food]++;
    }

    ksort($foodItems);
    $sortedFoodItems = array_keys($foodItems);

    $result = [];
    $header = array_merge(["Table"], $sortedFoodItems);
    $result[] = $header;

    ksort($tableOrders, SORT_NUMERIC);
    foreach ($tableOrders as $table => $foodCounts) {
        $row = [$table];
        foreach ($sortedFoodItems as $food) {
            $row[] = isset($foodCounts[$food]) ? strval($foodCounts[$food]) : "0";
        }
        $result[] = $row;
    }

    return $result;
}

$orders = [
    ["David", "3", "Ceviche"],
    ["Corina", "10", "Beef Burrito"],
    ["David", "3", "Fried Chicken"],
    ["Carla", "5", "Water"],
    ["Carla", "5", "Ceviche"],
    ["Rous", "3", "Ceviche"]
];

var_dump(displayTable($orders));

File: Python/Medium/Sorting/1418__Display_Table_of_Food_Orders_in_a_Restaurant.py
from collections import defaultdict


def displayTable(orders: list[list[str]]) -> list[list[str]]:
    food_items = set()
    table_orders = defaultdict(lambda: defaultdict(int))

    for _, table, food in orders:
        food_items.add(food)
        table_orders[table][food] += 1

    sorted_food_items = sorted(food_items)

    result = []
    result.append(["Table"] + sorted_food_items)

    for table in sorted(table_orders.keys(), key=int):
        row = [table] + [str(table_orders[table].get(food, 0)) for food in sorted_food_items]
        result.append(row)

    return result


orders = [["David", "3", "Ceviche"], ["Corina", "10", "Beef Burrito"], ["David", "3", "Fried Chicken"],
          ["Carla", "5", "Water"], ["Carla", "5", "Ceviche"], ["Rous", "3", "Ceviche"]]
print(displayTable(orders))
